core/src/main/java/hivemall/utils/collections/lists/DoubleArrayList.java

```diff
@@ -70,7 +70,7 @@ public DoubleArrayList add(@Nonnull double[] values) {
     private void expand(int max) {
         while (data.length < max) {
             final int len = data.length;
-            double[] newArray = new double[len * 2];
+            double[] newArray = new double[(len + 1) * 2];
```

Comment:
why this change has been made?

Reply:
See d7695d461056b21eab25465e015c582edc2b57ce

In case # of samples is less than 10, [`SmileExtUtils#sort` calls `DoubleArrayList(0)`](https://github.com/apache/incubator-hivemall/blob/7b61a3be8178fef40154f4009fef2989a9fd4c49/core/src/main/java/hivemall/smile/utils/SmileExtUtils.java#L190-L192), and zero-sized array list falls into infinite loop in the `expand` method as `newArray = new double[0 * 2]`. The change prevents this.

Reply:
hmm, this PR does not resolves a potential bug in expand. Returning array should be `>= max` and `max` should be `minCapacity` where `expand`'s argument is expected to be >=1.

https://github.com/karussell/fastutil/blob/master/src/it/unimi/dsi/fastutil/doubles/DoubleArrayList.java#L203
https://github.com/karussell/fastutil/blob/master/src/it/unimi/dsi/fastutil/doubles/DoubleArrays.java#L136

Let me fix this in another PR.